designs/oauth.rst

```diff
@@ -93,21 +93,21 @@ made based on other factors like group membership.
 
```

Comment:
How do kanidm roles (are we talking about the roles for the access control?) relate to scopes?

Reply:
In oauth2 a scope is a request that "does this account have the permissions to access this scope". It's a string blob effectively, but we need to make an access decision about that.

Consider we are asked for a user to have a scope of "email" which generally means "can this user read their own email?". In this case, we may consider implicitly granting this, since all users can read their own email.

But in other more complex cases, we may want to consider a resource server may request a scope unique to itself. IE a wiki instance may request "wiki_edit" as a scope from kanidm. We need a way to map that to "who get's this permission?". That means we need to have a way to express on the connected resource server, how to map that resource servers scopes, to groups or other access mechanisms in kani that reflect that.

Again, these need to be per resource server, because there may be duplicated scope strings between different resource servers. One could imagine a situation such as confluence which has a scope of "wiki_admin" that would be a different permission from a media wiki's "wiki_admin" scope. They are "resource server" specific permissions, so we need to then in kanidm, map those potentially to groups or other things. 

IE we could have a potentially future enhancement which is a way to allow a resource server with a scope of "trusted network" and that could potentially have a way to map an express to IP mapping rules that could come from the associated users auth token. But that would require further design and consideration to create that.